cmd/main.go

```diff
@@ -1,79 +1,201 @@
 
-	var httpPort int
-	flag.IntVar(&httpPort, "http_port", 8082, "The HTTP server port")
+		err := infradb.NewInfraDB(config.GlobalConfig.DBAddress, config.GlobalConfig.Database)
+		if err != nil {
+			log.Println("error in creating db", err)
+		}
+		go runGatewayServer(config.GlobalConfig.GRPCPort, config.GlobalConfig.HTTPPort)
 
-	var tlsFiles string
-	flag.StringVar(&tlsFiles, "tls", "", "TLS files in server_cert:server_key:ca_cert format.")
+		defer func() {
+			if err := infradb.Close(); err != nil {
+				log.Fatal(err)
+			}
+		}()
 
-	var redisAddress string
-	flag.StringVar(&redisAddress, "redis_addr", "127.0.0.1:6379", "Redis address in ip_address:port format")
+		switch config.GlobalConfig.Buildenv {
+		case "ipu":
+			gen_linux.Init()
+			ipu_linux.Init()
+			frr.Init()
+
+		case "ci":
+			gen_linux.Init()
+			ci_linux.Init()
+			frr.Init()
+		default:
+			log.Fatal(" ERROR: Could not find Build env ")
+		}
 
-	var frrAddress string
-	flag.StringVar(&frrAddress, "frr_addr", "127.0.0.1", "Frr address in ip_address format, no port")
+		// Create GRD VRF configuration during startup
+		if err := createGrdVrf(); err != nil {
+			log.Printf("Error in creating GRD VRF %+v\n", err)
+		}
+
+		runGrpcServer(config.GlobalConfig.GRPCPort, config.GlobalConfig.TLSFiles)
+	},
+}
 
-	flag.Parse()
+// initialize the cobra configuration and bind the flags
+func initialize() {
+	cobra.OnInitialize(initConfig)
 
-	// Create KV store for persistence
-	options := redis.DefaultOptions
-	options.Address = redisAddress
-	options.Codec = utils.ProtoCodec{}
-	store, err := redis.NewClient(options)
+	rootCmd.PersistentFlags().StringVarP(&config.GlobalConfig.CfgFile, "config", "c", "config.yaml", "config file path")
+	rootCmd.PersistentFlags().IntVar(&config.GlobalConfig.GRPCPort, "grpcport", 50151, "The gRPC server port")
+	rootCmd.PersistentFlags().IntVar(&config.GlobalConfig.HTTPPort, "httpport", 8082, "The HTTP server port")
+	rootCmd.PersistentFlags().StringVar(&config.GlobalConfig.TLSFiles, "tlsfiles", "", "TLS files in server_cert:server_key:ca_cert format.")
+	rootCmd.PersistentFlags().StringVar(&config.GlobalConfig.DBAddress, "dbaddress", "127.0.0.1:6379", "db address in ip_address:port format")
+	rootCmd.PersistentFlags().StringVar(&config.GlobalConfig.FRRAddress, "frraddress", "127.0.0.1", "Frr address in ip_address format, no port")
+	rootCmd.PersistentFlags().StringVar(&config.GlobalConfig.Database, "database", "redis", "Database connection string")
+
+	if err := viper.GetViper().BindPFlags(rootCmd.PersistentFlags()); err != nil {
+		log.Printf("Error binding flags to Viper: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+// initConfig read the config from file
+func initConfig() {
+	if config.GlobalConfig.CfgFile != "" {
+		viper.SetConfigFile(config.GlobalConfig.CfgFile)
+	} else {
+		// Search config in the default location
+		viper.AddConfigPath(configFilePath)
+		viper.SetConfigType("yaml")
+		viper.SetConfigName("config.yaml")
+	}
+	config.LoadConfig()
```

Comment:
Should the LoadConfig validate the file information that it is reading in case of errors in the information?  Should that be part of ValidateConfigs operation below?

Reply:
@atulpatel261194 will reply on this

Reply:
added error handling.